New `add_self_loops(g)` for hetero graphs

rbSparky opened the pull request:
As mentioned in PR #400, adding a new `add_self_loops(g)` function for hetero graphs would be helpful.

This PR does that. 

Will integrate this function to be used in `conv.jl`.
- [x] SGConv
- [x] GATConv (when #400 is merged)

CarloLucibello commented:
No need to reimplement everything. The new method `add_self_loops(hg)` should just call the `add_self_loops(hg, edge_t)` on each edge type. 

Then the docstring here 
https://github.com/CarloLucibello/GraphNeuralNetworks.jl/blob/5250c4dae953c70e6feae27f44df7d7f6a64eacc/src/GNNGraphs/transform.jl#L56
should be describe the additional behavior.

rbSparky commented:
> No need to reimplement everything. The new method `add_self_loops(hg)` should just call the `add_self_loops(hg, edge_t)` on each edge type.
> 
> Then the docstring here
> 
> https://github.com/CarloLucibello/GraphNeuralNetworks.jl/blob/5250c4dae953c70e6feae27f44df7d7f6a64eacc/src/GNNGraphs/transform.jl#L56
> 
> should be describe the additional behavior.

Changed function and changed docstring to include the new function and what it does

CarloLucibello commented on src/GNNGraphs/transform.jl:
```diff
@@ -40,18 +40,19 @@ end
-a second set of self-loops of the same type.
-
-If the graph has edge weights for edges of type `edge_t`, the new edges will have weight 1.
+- For a specific edge type (`edge_t`):
```
Let's not start a docstring with a bullet point. 
The change to the docstring can be minimal, just adding a paragraph at the end saying "If `edge_t` is not passed as argument, ..."

rbSparky replied:
Done

rbSparky commented:
Once this is merged I can make a PR for GATv2 hetero using this function
